docs/dynamic_publisher.md

```diff
@@ -0,0 +1,85 @@
+Publisher
+=========
+The DynamicPublisher block sends signals to a dynamic topic. The topic expression is evaulated for _each_ signal, and the resulting topic is used to publish. The entire signal must be JSON-serializable. 
+
+For more information on publishing to topics, view the [pub/sub docs.](https://docs.n.io/service-design-patterns/pub-sub.html)
+
+Properties
+----------
+- **topic**: Hierarchical topic expression to publish the incoming signals to. 
+
+Advanced Properties
+---
+
+- **ttl**: Time-to-live for each published **topic**. If no signals have been received for this **topic** within the **ttl** window, then the Publisher is eligible for removal/garbage-collection.
+
+Examples
+---
+
+##### Basic Usage
+
+You can use a DynamicPublisher block to publish a signal to a topic that is derived from the incoming signal. Consider the following `DynamicPublisher` configuration:
+
+```yaml
+topic: "airports.{{ $code }}.temperature"
```

Comment:
surely this should say `"airports.{{ $location }}.temperature"`, judging from L28

Reply:
oops, meant to do it the other way when I refactored/updated the docs. Fixed